File: app/src/main/java/com/kabirnayeem99/paymentpaid/data/db/entities/Work.java
package com.kabirnayeem99.paymentpaid.data.db.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "works_db_table")
public class Work {
    @PrimaryKey(autoGenerate = true)
    private int id;
    @ColumnInfo(name = "work_name")
    private final String name;
    @ColumnInfo(name = "submission_date")
    private final String date;
    @ColumnInfo(name = "account_month")
    private final int month;
    @ColumnInfo(name = "account_year")
    private final int year;
    @ColumnInfo(name = "payment")
    private final int payment;
    @ColumnInfo(name = "student_name")
    private final String studentName;

    public Work(String name, String date, int month, int year, int payment, String studentName) {
        this.name = name;
        this.date = date;
        this.month = month;
        this.year = year;
        this.payment = payment;
        this.studentName = studentName;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public String getDate() {
        return date;
    }

    public int getMonth() {
        return month;
    }

    public int getYear() {
        return year;
    }

    public int getPayment() {
        return payment;
    }

    public String getStudentName() {
        return studentName;
    }

    @Override
    public String toString() {
        return "Work{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", date='" + date + '\'' +
                ", month=" + month +
                ", year=" + year +
                ", payment=" + payment +
                ", studentName='" + studentName + '\'' +
                '}';
    }
}

File: app/src/main/java/com/kabirnayeem99/paymentpaid/data/db/entities/Work.kt
package com.kabirnayeem99.paymentpaid.data.db.entities

import android.os.Parcelable
import android.util.Log
import com.google.firebase.firestore.DocumentSnapshot
import kotlinx.android.parcel.Parcelize
import java.lang.Exception

/**
 * Represents a work
 * @property name the name of the work, in a format like student_name-work_name_shortcut
 * @property day submission day of the work
 * @property month the month of the work
 * @property year the year of the work
 */
@Parcelize
data class Work(
        var documentId: String?,
        var name: String,
        var day: Long,
        var month: Long,
        var year: Long,
        var payment: Long,
        var studentName: String
) : Parcelable {


    companion object {
        private const val TAG = "Work"
        fun DocumentSnapshot.toWork(): Work? {
            try {
                val name = getString("name")!!
                val day = getLong("day")!!
                val month = getLong("month")!!
                val year = getLong("year")!!
                val payment = getLong("payment")!!
                val studentName = getString("studentName")!!

                return Work(id, name, day, month, year, payment, studentName)
            } catch (e: Exception) {

                Log.e(TAG, "toWork: error converting work $e")
                //todo: implement crash analytics later
//                FirebaseCrashlytics.getInstance().log("Error converting user profile")
//                FirebaseCrashlytics.getInstance().setCustomKey("userId", id)
//                FirebaseCrashlytics.getInstance().recordException(e)
                return null
            }
        }

        fun createWork(){

        }
    }
}
